### DISCUSSION TITLE
Is it possible to use TinyUSB on Android?

### DISCUSSION THREAD

**Discussion by FrancoisDn** [hathach/tinyusb#2138] [Q&A] [answered]
Hello,
I have a very stupid question...
Is it possible to used TinyUSB on Android device?

I would like to read input video through USB (OTG), using UVC. I see that there is an example to do that, but I'm block at the very beginning, I'm not able to compile the project, I do not know which "board" select to compile on Linux (x86/64) or (Android) for supporting all USB devices compatible with UVC.

Is it possible to compile the library (and how?) to support most devices on Linux and Android?

**Comment by hathach** [comment, marked as answer]
I don't think tinyusb would be able to run on Linux as the current state. You are better using Linux  usb driver for this purpose.
